### PR TITLE
Add an analytic data PolarMagnetizedFmDisk

### PR DESCRIPTION
**Pull request by jyoo1042**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp [hunk 1]
```diff
@@ -23,9 +23,10 @@ void test_tag_retrieval(const Solution& solution, const Coords& coords,
   tmpl::for_each<TagsList>([&](const auto tag_v) {
     using tag = tmpl::type_from<decltype(tag_v)>;
     const auto single_var = solution.variables(coords, tmpl::list<tag>{});
-    CHECK(tuples::get<tag>(single_var) == tuples::get<tag>(vars_from_all_tags));
-    CHECK(tuples::get<tag>(single_var) ==
-          tuples::get<tag>(vars_from_all_tags_reversed));
+    CHECK_ITERABLE_APPROX(tuples::get<tag>(single_var),
+                          tuples::get<tag>(vars_from_all_tags));
```
Comment (wthrowe) [inline tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp:27]: Why this change?

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_PolarMagnetizedFmDisk.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+void test_move() {
+  grmhd::AnalyticData::PolarMagnetizedFmDisk disk(
+            grmhd::AnalyticData::MagnetizedFmDisk(1.3, 0.345, 6.123, 14.2,
+                                                  0.065, 1.654, 0.42, 85.0, 6),
+            domain::CoordinateMaps::SphericalTorus(2.5, 3.6, 0.9, 0.7));
+  grmhd::AnalyticData::PolarMagnetizedFmDisk disk_copy(
+            grmhd::AnalyticData::MagnetizedFmDisk(1.3, 0.345, 6.123, 14.2,
+                                                  0.065, 1.654, 0.42, 85.0, 6),
+            domain::CoordinateMaps::SphericalTorus(2.5, 3.6, 0.9, 0.7));
+  test_move_semantics(std::move(disk), disk_copy);  //  NOLINT
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_PolarMagnetizedFmDisk.cpp]: Indicate what warning is being silenced by the NOLINT as `// NOLINT(warning-name)` or `// NOLINTNEXTLINE(warning-name)`, where `warning-name` is from the error message and will look something like `performance-move-const-arg`.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_PolarMagnetizedFmDisk.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticData.GrMhd.PolarMagFmDisk",
+                  "[Unit][PointwiseFunctions]") {
+  test_create_from_options();
+  test_serialize();
+  test_move();
+  test_variables(std::numeric_limits<double>::signaling_NaN());
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_PolarMagnetizedFmDisk.cpp:134]: Also test with DataVector.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace {
+const std::array<double, 3> t2a(const tnsr::I<double, 3>& tens);
+const tnsr::I<double, 3> a2t(const std::array<double, 3>& arr);
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: No need to declare these here, since they are defined immediately following.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      for (size_t j = 0; j < 3; ++j) {
+        CAPTURE(j);
+        const auto numerical0 = numerical_derivative(
+            [&i, &partial_torus](const std::array<double, 3>& x) {
+              const auto x_tnsr = a2t(x);
+              return partial_torus(x_tnsr).get(i);
+            },
+            test_point, j, 1e-3);
+
+        const auto numerical1 = numerical_derivative(
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: Use more descriptive names for these variables.  Maybe `numerical_jacobian` and `numerical_inverse_jacobian` or something.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  CHECK(full_torus !=
+        grmhd::AnalyticData::SphericalTorus(r_min, r_max + 0.1, phi_max));
+  CHECK(full_torus !=
+        grmhd::AnalyticData::SphericalTorus(r_min, r_max, phi_max + 0.1));
+
+  CHECK(not full_torus.is_identity());
+  CHECK(not partial_torus.is_identity());
+
+  {
+    auto num_approx = Approx::custom().epsilon(1.0e-9).scale(1.0);
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: What does "num" indicate?  Don't abbreviate variable names too much.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace {
+const std::array<double, 3> t2a(const tnsr::I<double, 3>& tens);
+const tnsr::I<double, 3> a2t(const std::array<double, 3>& arr);
+
+const std::array<double, 3> t2a(const tnsr::I<double, 3>& tens) {
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: `#include "DataStructures/Tensor/Tensor.hpp"`

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  std::array<double, 3> arr{};
+  for (size_t i = 0; i < 3; i++) {
+    gsl::at(arr,i) = tens.get(i);
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp:22]: `#include "Utilities/Gsl.hpp"`

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+#include "PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace {
+const std::array<double, 3> t2a(const tnsr::I<double, 3>& tens);
+const tnsr::I<double, 3> a2t(const std::array<double, 3>& arr);
+
+const std::array<double, 3> t2a(const tnsr::I<double, 3>& tens) {
+  std::array<double, 3> arr{};
+  for (size_t i = 0; i < 3; i++) {
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp:21]: `#include <cstddef>`

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+    gsl::at(arr,i) = tens.get(i);
+  }
+  return arr;
+}
+
+const tnsr::I<double, 3> a2t(const std::array<double, 3>& arr) {
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: Remove `const` from return type.  No reason to restrict what the caller can do with the temporary.  Also on `t2a`.

File: src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt [hunk 11]
```diff
@@ -140,6 +140,12 @@ add_grmhd_executable(
   ""
   )
 
+add_grmhd_executable(
+  PolarMagnetizedFmDisk
+  grmhd::AnalyticData::PolarMagnetizedFmDisk
+  "${LIBS_TO_LINK}"
```
Comment (wthrowe) [inline src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt]: Remove `${LIBS_TO_LINK}` in this commit instead of the next one.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp (hunk 1 above)
Comment (jyoo1042) [reply tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp:27]: test_tag_retrieval was giving an error when using CHECK.

Comment (wthrowe) [reply tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp:27]: 1. There are no other changes in this commit, so probably this should be in one of the later commits.
2. Do you understand why?  It might be fine, but I don't see an obvious reason for them to differ,

Comment (wthrowe) [reply tests/Unit/Helpers/PointwiseFunctions/AnalyticData/TestHelpers.hpp:27]: In offline discussion we decided this change is fine squashed into a later commit.  It appears to be optimization related, and this class has more interaction between fetched variables than most of the others so that's not terribly unexpected.

**Review by wthrowe — DISMISSED** [review]
I'm happy with this, but since I was involved in writing some of the code I shouldn't be the sole reviewer.  @nilsdeppe, I think you were willing to do the follow-up?

**Comment by nilsdeppe** [comment]
Yep! I might not have time until later next week, but I'm happy to take a look, hopefully sooner :D

File: src/Domain/CoordinateMaps/CoordinateMap.tpp [hunk 12]
```diff
@@ -125,9 +125,10 @@ bool CoordinateMap<SourceFrame, TargetFrame,
 
 template <typename SourceFrame, typename TargetFrame, typename... Maps>
 void CoordinateMap<SourceFrame, TargetFrame, Maps...>::check_functions_of_time(
-    [[maybe_unused]] const std::unordered_map<
+    const std::unordered_map<
         std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
         functions_of_time) const {
+(void) functions_of_time;
```
Comment (nilsdeppe) [inline src/Domain/CoordinateMaps/CoordinateMap.tpp]: What do you have against `[[maybe_unused]]`?

File: src/PointwiseFunctions/AnalyticData/GrMhd/PolarMagnetizedFmDisk.hpp [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd::AnalyticData {
+/*!
+ * \brief Magnetized fluid disk orbiting a Kerr black hole in the usual
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/PolarMagnetizedFmDisk.hpp]: maybe clarify what "usual" means?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Couple minor things, please squash right away

File: src/Domain/CoordinateMaps/CoordinateMap.tpp (hunk 12 above)
Comment (jyoo1042) [reply src/Domain/CoordinateMaps/CoordinateMap.tpp]: I got a compiler warning even with `[[maybe_unused]]`.
But, I can undo this change because PolarMagnetizedFmDisk.cpp no longer uses this file.

File: src/PointwiseFunctions/AnalyticData/GrMhd/PolarMagnetizedFmDisk.hpp [hunk 14]
```diff
@@ -34,9 +33,9 @@ class er;
 
 namespace grmhd::AnalyticData {
 /*!
- * \brief Magnetized fluid disk orbiting a Kerr black hole in the usual
+ * \brief Magnetized fluid disk orbiting a Kerr black hole in the Kerr-Schild
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/PolarMagnetizedFmDisk.hpp:36]: This should go in the first commit.

File: src/Domain/CoordinateMaps/CoordinateMap.tpp (hunk 12 above)
Comment (nilsdeppe) [reply src/Domain/CoordinateMaps/CoordinateMap.tpp]: Yes, please undo since it's not needed

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — APPROVED** [review]
